code/datums/components/chem_booster.dm

```diff
@@ -277,6 +279,23 @@
 		meds_beaker.reagents.trans_to(wearer, 30)
 	setup_bonus_effects()
 
+///Updates the module on the armor to glow or not
+/datum/component/chem_booster/proc/update_module_icon(var/on_off)
+	var/obj/item/clothing/suit/modular/parent_armor = parent
+	var/obj/item/armor_module/module/chemsystem/chemsystem_icon = parent_armor.attachments_by_slot[ATTACHMENT_SLOT_MODULE]
+	if(isnull(chemsystem_icon))
+		return
+	if(on_off)
+		chemsystem_icon.icon_state = "mod_chemsystem_active"
+		chemsystem_icon.item_state = "mod_chemsystem_active_a"
+		chemsystem_icon.update_icon()
+		parent_armor.update_icon()
+		return
+	chemsystem_icon.icon_state = "mod_chemsystem"
+	chemsystem_icon.item_state = "mod_chemsystem_a"
+	chemsystem_icon.update_icon()
```

Comment:
This isn't updating icons correctly, it should be done inside the actual vali module, and it will need to be set to respect icon overrides.